lib/rucio/db/sqla/types.py

```diff
@@ -93,9 +93,9 @@ def process_result_value(self, value, dialect):
         if value is None:
             return value
 
-        if value == '1':
+        if value.lower() in ['1', 'true']:
             return True
-        elif value == '0':
+        elif value.lower() in ['0', 'false']:
             return False
```

Comment:
Sometimes 0 /1 are simple numerical values like tier 1. What about dropping 0 / 1 to use only true/false for booleans  ? I t can help to have a clear type specification from the clients.  For the db non supporting the boolean type, one additional field/column/etc telling if the value is a boolean or a numerical value can be added to help the json serialisation imo.

Reply:
I messed up the commit somehow, that's what I wanted to do. FOr now we have to keep '0','1' as backward compatible, but store only 'true'/'false' as string. For the type, we know that already, since they are set by the SQLA model.

Reply:
Like before: Python Boolean True. IMHO this is wrong conceptually, but that's what we have right now.

We need to gradually migrate installations over to the new 'true'/'false' behaviour, once this code is in place.

Reply:
I think this should be a feature, looks like a potential impactful change.
I wonder if it is not worth to introduce an additional column `value_type` right away and not trick around with some intermediate states?

Reply:
`value_type` won't really help, will it? unless we expect users to always define the type when setting the key/value pair, which is a bit weird.

but i agree on making this a feature. let me cancel this one then so it doesn't get too confusing.

Reply:
If it comes from the API, we can extract the type. From the CLI we probably would need a `--boolean` parameter.

Reply:
Isn't this how it should be:

- strings `true`/`false` are interpreted as boolean
- any number is a number (0 and 1 not interpreted as boolean, that was the mistake we made)
- anything else is string

No need for a parameter that way, and IMHO much more intuitive.

Reply:
using true/false like in json should be good enough imo. With pseudo commands: `set attr attrname true`, `set attr attrname false`, `set attr name` can get `true` by default.

Reply:
ok, so we agree :-D

Reply:
Yes that sounds good, I have to check the RSE-Expression though as well, this might impact them. (e.g. CERN-PROD_DATADISK matches against a boolean attribute, need to check if this is done correctly)

Reply:
RSE Expression continue to work as before only on MySQL. I'll address this in my patch.